content/scripts/build.js

```diff
@@ -1043,22 +1158,37 @@ class Builder {
     // like `<--#Compat('foo.bar')--> and then replace it here in the
     // post-processing instead.
 
-    // // REAL
-    // const rawHtml = fs.readFileSync(path.join(folder, "index.html"), "utf8");
-    // hasher.update(rawHtml);
-    // const renderedHtml = this.renderHtml(rawHtml, metadata);
-    // FAKE (NOTE, the docHash check stuff needs to happen BEFORE executing renderHtml)
-    // const rawHtml = fs.readFileSync(path.join(folder, "raw.html"), "utf8");
-    const renderedHtml = fs.readFileSync(
-      path.join(folder, "index.html"),
-      "utf8"
-    );
-    hasher.update(renderedHtml);
+    const rawHtml = fs.readFileSync(path.join(folder, "index.html"), "utf8");
+
+    let renderedHtml;
+
+    // When 'source.htmlAlreadyRendered' is true, it simply means that the 'index.html'
+    // is already fully rendered HTML.
+    if (source.htmlAlreadyRendered) {
+      renderedHtml = rawHtml;
+    } else {
+      // Errors while KS rendering will be accumulated during the build
+      // and reported at the end.
+      renderedHtml = await this.renderMacros(rawHtml, metadata);
+      if (
+        this.options.flaws === FLAWS_LEVELS.ERROR &&
+        this.reportMacroRenderingFlaws()
+      ) {
+        // Report and exit immediately on the first document with flaws.
+        process.exit(1);
```

Comment:
This is going to need some work. Somewhere :)
If you run `node content build --flaws=error` you want it to die quickly. But if you're using the watcher or the `server` you don't want that to die. The `server` for example would need to be something like:
```javascript
try {
    builder.start(specificFolders);
} catch(err) {
   res.status(500).send(err.toString())
}
```
...or something. We can refine this later.

Reply:
I agree. Currently, with what I have, the server reports flaws (since they're not yet reported via your flaws system on the actual page) but never dies (see https://github.com/mdn/stumptown-renderer/blob/9cef117a9de52899b634d367e641b1ed404f4138/server/index.js#L120), and the watcher behaves as configured from the command line (or environment) (see https://github.com/mdn/stumptown-renderer/blob/9cef117a9de52899b634d367e641b1ed404f4138/content/scripts/build.js#L761).